src/python/pants/bin/pants_runner.py

```diff
@@ -79,6 +78,10 @@ def run(self, start_time: float) -> ExitCode:
             stdout_fileno=stdout_fileno,
             stderr_fileno=stderr_fileno,
         ):
+            # N.B. Inlining these imports speeds up the python thin client run, and avoids importing
+            # engine types until after the runner has had a chance to set PANTS_BIN_NAME.
+            from pants.bin.local_pants_runner import LocalPantsRunner
+            from pants.bin.remote_pants_runner import RemotePantsRunner
```

Comment:
Can this be moved into the `if` statement? If we aren't using Pantsd, no reason to import it.